velox/connectors/hive/HiveConnectorSplit.h

```diff
@@ -33,7 +33,7 @@ namespace facebook::velox::connector::hive {
 struct HiveBucketConversion {
   int32_t tableBucketCount;
   int32_t partitionBucketCount;
-  std::vector<std::unique_ptr<HiveColumnHandle>> bucketColumnHandles;
+  std::vector<std::shared_ptr<HiveColumnHandle>> bucketColumnHandles;
```

Comment:
curiously why do we need to change it to shared_ptr?

Reply:
It's because Velox's JSON serde always deserializes JSON to `shared_ptr` objects.

Could refer to [this API](https://github.com/facebookincubator/velox/blob/82e00de0eabea540cdec3d9b5bff6d2c2260b0fa/velox/common/serialization/Serializable.h#L171-L186) and [this API](https://github.com/facebookincubator/velox/blob/82e00de0eabea540cdec3d9b5bff6d2c2260b0fa/velox/connectors/hive/TableHandle.h#L117) as examples.

So we cannot get `unique_ptr<HiveColumnHandle>` from the deserialization API without changing the serde internals.